zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java

```diff
@@ -159,6 +159,16 @@ public static ClientAuth fromPropertyValue(String prop) {
     private String sslClientAuthProperty = getConfigPrefix() + "clientAuth";
     private String sslHandshakeDetectionTimeoutMillisProperty = getConfigPrefix() + "handshakeDetectionTimeoutMillis";
 
+    /**
+     * Config properties for ZNode group ACL features
+     */
+    public static final String ZNODE_GROUP_ACL_CONFIG_PREFIX = "zookeeper.ssl.znodeGroupAcl.";
+    // Enables/disables whether znodes created by auth'ed clients
+    // should have ACL fields populated with the client Id given by the authentication provider.
+    // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
+    public static final String SET_X509_CLIENT_ID_AS_ACL =
```

Comment:
Besides this configuration, I'm thinking we should have another configuration property that controls the on/off of the entire znode group acl feature, which functions like a main switch, what do you think?

Reply:
Let's hold off on this until we get some more clarity on the real value of add such a config.